04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,67 @@
+const Product = require('../models/product');
+
+const getAllProductsStatic = async (req, res) => {
+  const products = await Product.find({ price: { $gt: 30 } })
+    .select('name price')
+    .limit(10)
+    .skip(5);
+  res.status(200).json({ products, nbHits: products.length });
+};
+
+const getAllProducts = async (req, res) => {
+  const { featured, company, name, sort, fields, numericFilters } = req.query;
+  const queryObject = {};
+  if (featured) {
+    queryObject.featured = featured === 'true' ? true : false;
+  }
+  if (company) {
+    queryObject.company = company;
+  }
+  if (name) {
+    queryObject.name = { $regex: name, $options: 'i' };
+  }
+
+  if (numericFilters) {
+    const operatorMap = {
+      '>': '$gt',
+      '>=': '$gte',
+      '=': '$eq',
+      '<': '$lt',
+      '<=': '$lte',
+    };
+    const regEx = /\b(>|>=|=|<|<=)\b/g;
+    let filters = numericFilters.replace(
+      regEx,
+      (match) => `-${operatorMap[match]}-`
+    );
```

Comment:
Hm, I'm not sure that there's enough information to understand _why_ we want to do this without more context? Why do we want to do this? It looks like we're replacing `$gt` with `-$gt-`, for example?

Reply:
We replace operators: ```>, >=, =, <, <= ``` to ```-```(hyphen) that we use as a delimiter to split the string of filters into individual filter expressions which can be used to construct a more complex query to search a MongoDB database.